packages/shylu/shylu_dd/frosch/test/Overlap/main.cpp

```diff
@@ -361,35 +420,96 @@ int main(int argc, char *argv[])
+        double adapt_end = MPI_Wtime();
+        if (Comm->getRank()==0) cout << "The code block for thes caling and setup of solveOp took "<< adapt_end-adapt_start<<"seconds"<<endl;
 
-        Comm->barrier(); if (Comm->getRank()==0) cout << "######################\n# Thyra PrepForSolve #\n######################\n" << endl;
+        // Comm->barrier(); if (Comm->getRank()==0) cout << "\n#########\n# Solve #\n#########" << endl;
+        SolveStatus<SC> status =
+        solve<SC>(*solveOp, Thyra::NOTRANS, *thyraBModified, thyraXModified.ptr());
+        FROSCH_ASSERT(status.solveStatus==SOLVE_STATUS_CONVERGED, "Solver didn't converge");
+        //Comm->barrier(); if (Comm->getRank()==0) cout << "the error is: - for the new system:" << thyraB->col(0)->norm_1() << " for the olds system " << thyraB->col(0)->norm_1()<<std::endl;
 
-        RCP<LinearOpWithSolveFactoryBase<SC> > lowsFactory =
-        linearSolverBuilder.createLinearSolveStrategy("");
+        // Comm->barrier(); if (Comm->getRank()==0) cout << "###########################\n# AfterSolve #\n###########################" << endl;
+        thyraX = thyraXModified->clone_mv();
+        froschLinearOp->afterSolve(thyraX.ptr());
 
-        lowsFactory->setOStream(out);
-        lowsFactory->setVerbLevel(VERB_HIGH);
+        double end = MPI_Wtime();
+        if (Comm->getRank()==0) cout << "Solving the system took " << end - start << " seconds to run." << endl;
 
-        Comm->barrier(); if (Comm->getRank()==0) cout << "###########################\n# Thyra LinearOpWithSolve #\n###########################" << endl;
+        // check if it solves the original problem:
+        K_thyra->apply(EOpTransp::NOTRANS,*thyraX, thyraB.ptr(),1.0,-1.0);
+        K_thyra->apply(EOpTransp::NOTRANS,*thyraXModified, thyraBModified.ptr(),1.0,-1.0);
+        SC residualModifiedSystem=norm(*(thyraBModified->col(0)));
+        SC residualOriginalSystem=norm(*(thyraB->col(0)));
 
-        RCP<LinearOpWithSolveBase<SC> > lows =
-        linearOpWithSolve(*lowsFactory, K_thyra);
 
-        Comm->barrier(); if (Comm->getRank()==0) cout << "\n#########\n# Solve #\n#########" << endl;
-        SolveStatus<SC> status =
-        solve<SC>(*lows, Thyra::NOTRANS, *thyraB, thyraX.ptr());
+        SC errorBefore = norm(*(thyraXExact->col(0)));
+        update(-1., *thyraX, thyraXExact.ptr());
+        //auto xpetravec = FROSch::toXpetra<SC,LO,GO,NO>(thyraXExact.ptr()); // Teuchos::RCP<Thyra::MultiVectorBase<SC>>
+        //output(xpetravec, "xExact", 0);
+
+        SC errorAfter = norm(*(thyraXExact->col(0)));
 
+        Comm->barrier(); if (Comm->getRank()==0) cout << "the absolut residual is: - for the Modified system: "<<residualModifiedSystem << " - for the original system: "<< residualOriginalSystem <<std::endl;
+        Comm->barrier(); if (Comm->getRank()==0) cout << "the error to the exact solution is: "<< errorAfter << ". Reduced from: " << errorBefore << std::endl;
```

Comment:
I did not go through all the new lines, but I was wondering how much of this is actually "needed" for using your new implementation?